src/utilities/include/public.h

```diff
@@ -292,17 +292,17 @@ struct Box {
   /**
    * Does this box overlap the one given (including equality)?
    */
-  bool DoesOverlap(const Box& box) const {
-    return min.x <= box.max.x && min.y <= box.max.y && min.z <= box.max.z &&
-           max.x >= box.min.x && max.y >= box.min.y && max.z >= box.min.z;
+  inline bool DoesOverlap(const Box& box) const {
+    return (min.x <= box.max.x) & (min.y <= box.max.y) & (min.z <= box.max.z) &
+           (max.x >= box.min.x) & (max.y >= box.min.y) & (max.z >= box.min.z);
```

Comment:
Do you not want to short-circuit the comparisons?

Reply:
ah I was modifying this for something else, forgot to remove it
I was trying to see if the short-circuiting is causing branch misprediction and stuff, but it seems that there is no significant performance implication